### ISSUE TITLE
Resume fine-tuning using checkpoints

### ISSUE THREAD

**Issue by main-voice** [2U1/Qwen2-VL-Finetune#94]
Hi,
I saw you will resume the training when detecting the checkpoints:

```python
if list(pathlib.Path(training_args.output_dir).glob("checkpoint-*")):
    trainer.train(resume_from_checkpoint=True)
else:
    trainer.train()
```

And when you load the dataset, you will load the json file and initialize a SupervisedDataset, but how do you sync the dataset index with the training progress? Or this is handled by the trainer in transformers package?

**Comment by 2U1** [comment]
The states are saved in the trainer_state.json when saving checkpoint.

**Comment by main-voice** [comment]
So, will it load the model from the latest checkpoint, calculate the data already trained on (batch size * steps), and then continue training from the remaining data?

**Comment by 2U1** [comment]
Yes thats right.

**Comment by main-voice** [comment]
Then I guess the script doesn't support shuffle the dataset right? Because you can't remember the dataset distribution of the last training.

**Comment by 2U1** [comment]
No they are saved in the rng_state.pth.
Huggingface trainer reads the state from the file and resumes it.
